net-mgmt/pfSense-pkg-zabbix-agent/files/usr/local/pkg/zabbix-agent.inc

```diff
@@ -55,15 +55,17 @@ function validate_input_zabbix_agent($post, &$input_errors) {
 		}
 
 		if ($post['listenip'] != '') {
-			if (!is_ipaddr_configured($post['listenip']) && !preg_match("/(127.0.0.1|0.0.0.0)/", $post['listenip'])) {
-				$input_errors[] = "'Listen IP' is not a configured IP address.";
+			$listenips = explode(",", $post['listenip']);	
+			foreach ($listenips as $listip) {
+				$listip = trim($listip);
+				if (!is_ipaddr_configured($listip) && !preg_match("/(127.0.0.1|0.0.0.0|::1|::)/", $listip)) {
+					$input_errors[] = "'Listen IP' is not a configured list of IP addresses.";
```

Comment:
Doing the test this way will repeat the same error multiple times if more than one entry fails the test. It should probably  Set a flag variable then break out of the loop on the first error, then print the error once.

Alternately, it could print the failed value in the message and print an error for each failed value. Encode the value in output for safety (e.g. `htmlspecialchars()`).

Reply:
+ printing error for each field
+ extra validation checks